omniduct/databases/sqlalchemy.py

```diff
@@ -27,6 +27,7 @@ def _init(self, dialect=None, driver=None, database=''):
         self.driver = driver
```

Comment:
@matthewwardrop re: lazy prep

this diff went through several iterations since snowflake uses a strange bit of logic when creating a sqlalchemy URI https://github.com/snowflakedb/snowflake-sqlalchemy/blob/master/base.py#L389 

During that iteration at one point I didn't have a port set, and changed the init here to 

```
assert self.remote is None or self._port is not None
```
above.

The call to `self.remote` is a prepared field and thus when initializing it's also prepared and since preparing ALSO checks the cache (which was not yet loaded), we hit a key error in the registry. I'm not sure this is a bug per se but if we do want to make some prepared fields are accessible during initialization, then we need to be careful about that kind of thing.